Philips.CodeAnalysis.MaintainabilityAnalyzers/Readability/AvoidMultipleConditionsOnSameLineAnalyzer.cs

```diff
@@ -0,0 +1,127 @@
+﻿// © 2023 Koninklijke Philips N.V. See License.md in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Philips.CodeAnalysis.Common;
+
+namespace Philips.CodeAnalysis.MaintainabilityAnalyzers.Readability
+{
+	/// <summary>
+	/// Report when a multi line condition statement (if or ?), does not include a newline on its logical operators.
+	/// </summary>
+	[DiagnosticAnalyzer(LanguageNames.CSharp)]
+	public class AvoidMultipleConditionsOnSameLineAnalyzer : DiagnosticAnalyzer
+	{
+		/// <summary>
+		/// Diagnostic Id for this analyzer.
+		/// </summary>
+		private const string Title = "Divide multiline conditions on the logical operators.";
+		private const string Message =
+			"Divide multiline conditions around line {0} such, that the logical operators are at the end " +
+			"of the line.";
+		private const string Description = "Divide long conditions in consistent way";
+		private const string Category = "Shoscar";
+
+		private static readonly DiagnosticDescriptor Rule =
+			new (
+				Helper.ToDiagnosticId(DiagnosticIds.AvoidMultipleConditionsOnSameLine),
+				Title,
+				Message,
+				Category,
+				DiagnosticSeverity.Warning,
+				isEnabledByDefault: true,
+				description: Description);
+
+		/// <summary>
+		/// <inheritdoc cref="DiagnosticAnalyzer.SupportedDiagnostics"/>
+		/// </summary>
+		public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics =>
+			ImmutableArray.Create(Rule);
+
+		/// <summary>
+		/// <inheritdoc/>
+		/// </summary>
+		public override void Initialize(AnalysisContext context)
+		{
+			context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+			context.EnableConcurrentExecution();
+			context.RegisterSyntaxNodeAction(AnalyzeNode, SyntaxKind.LogicalAndExpression);
+			context.RegisterSyntaxNodeAction(AnalyzeNode, SyntaxKind.LogicalOrExpression);
+		}
+
+		private void AnalyzeNode(SyntaxNodeAnalysisContext context)
+		{
+			GeneratedCodeDetector generatedCodeDetector = new();
+			if (generatedCodeDetector.IsGeneratedCode(context))
+			{
+				return;
```

Comment:
You presumably aren't getting code coverage of this return condition because of the configuration flags earlier. Based on the findings of the GeneratedCodeDetector investigation, I think we can evolve toward eliminating it altogether and rely on the "new" flags instead. If you want Unit Test this condition, you can follow the pattern from RedundantSwitchAnalyzer (constructor overload and TestClass inheritance.

Reply:
I like the GeneratedCodeDetector, as it opens up the possibility to offer configuration to the user of generated code. If the generated code _forgets_ to write the GeneratedCodeAttribute, the user has somewhere to go.